### DISCUSSION TITLE
Run as an Android application

### DISCUSSION THREAD

**Discussion by furkanturkmn** [bluenviron/mediamtx#1757] [General]
I was trying to integrate the server into an android app by generating a library and starting the server through JNI call which is exposed over cgo. Inorder to build the server as a library I was using; CGO_ENABLED=1 GOOS=android GOARCH=arm64 CC=/home/dev/Android-SDK/ndk/25.1.8937393/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android33-clang go build -buildmode=c-shared -o libfoo.so main.go as a result the server gets called but somehow it crashes and the android subsystem repeatedly tries to reinitialize the server library. Do you have any clue what the problem might be?

**Comment by aler9** [comment]
Hello, my suggestion is to compile the server without cgo, embed it into the android app as a static file and start it by using Java's `ProcessBuilder()`.
The server doesn't expose any internal API, therefore JNI is useless, expecially if it leads to crashes.

[a comment by an automated account is not shown]
